src/app/Processor/ProcessorEdit/SourceEdit/SourceEdit.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  TextInput,
+} from "@patternfly/react-core";
+
+interface SourceEditProps {
+  source?: EventSource;
+  onChange: (source: EventSource) => void;
+}
+
+const SourceEdit = (props: SourceEditProps) => {
+  const { source, onChange } = props;
+  const [type, setType] = useState(source?.type ?? "");
+  const [parameters, setParameters] = useState(source?.parameters ?? {});
+
+  const updateType = (type: string) => {
+    setType(type);
+    onChange({
+      ...source,
+      type: type,
+      parameters,
+    });
+  };
+
+  const updateParameters = (parameters: EventSource["parameters"]) => {
+    setParameters(parameters);
+    onChange({
+      ...source,
+      type: type,
+      parameters,
+    });
+  };
+
+  const { t } = useTranslation(["openbridgeTempDictionary"]);
+
+  const sourceTypes = [
+    {
+      value: "",
+      label: t("processor.selectSource"),
+      isPlaceholder: true,
+    },
+    { value: "demoSource", label: "Demo Source", isPlaceholder: false },
+  ];
+
+  useEffect(() => {
+    if (source) {
+      setType(source.type);
+      setParameters(source.parameters);
+    }
+  }, [source]);
+
+  return (
+    <>
+      <FormGroup
+        fieldId={`source-type`}
+        label={t("processor.sourceType")}
+        isRequired={true}
+      >
+        <FormSelect
+          id={`source-type`}
+          aria-label={t("processor.sourceType")}
+          isRequired={true}
+          value={type}
+          onChange={(type) => updateType(type)}
+        >
+          {sourceTypes.map((option, index) => (
+            <FormSelectOption
+              key={index}
+              value={option.value}
+              label={option.label}
+              isPlaceholder={option.isPlaceholder}
+            />
```

Comment:
I don't see a great improvement honestly. It would be another component to test/maintain/migrate/be-aware-of.
I appreciate your suggestion but I don't think it's worth doing it.